.github/workflows/docusaurus.yml

```diff
@@ -0,0 +1,44 @@
+name: docusaurus
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Build Docusaurus
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          cache: "yarn"
+          cache-dependency-path: "app"
+
+      - name: Set up Yarn
+        working-directory: ./doc
+        run: |
+          corepack enable
+          corepack prepare yarn@4.3.1 --activate
+
+      - name: Install dependencies
+        working-directory: ./doc
+        run: yarn install --frozen-lockfile
```

Comment:
Why do we need `--frozen-lockfile`?

Reply:
The --frozen-lockfile flag is used with Yarn to ensure that the yarn.lock file is not modified during the installation process. This is particularly useful in CI/CD pipelines to guarantee that the exact versions of dependencies specified in the yarn.lock file are installed, ensuring consistency and reproducibility across different environments.